### PR TITLE
[stable/efs-provisioner] Add nodeselector, affinity, podAnnotations

### PR DESCRIPTION
**Pull request by SweetOps**
#### What this PR does / why we need it:

Adds `nodeSelector`, `affinity` to schedule pods in specific nodes and  `podAnnotations` to be able to annotate deployment. 

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

**Comment by whereisaaron** [comment]
ref #9127 patch which is needed for newer k8s RBAC rules.

**Comment by whereisaaron** [comment]
This of one of four PR's to add the same selector and/or pod annotations functionality to this chart (#10205, #10088, #10013, #9253), stuck in the backlog of unloved PRs 😢

**Comment by SweetOps** [comment]
@whereisaaron , may be more PRs will help to speed up merging :) . You can close this PR, it nvm for me which PR will be merged - me just need that functionality

**Comment by whereisaaron** [comment]
Maybe @SweetOps! 😀 I wasn’t making any suggestion as to which PRs should close or merge, just noting the slow review of PR’s leads to wasteful duplication of effort. You had to do the same work to improve chart as others before you, instead of being able to share your work in an easy, timely manner.

A more distributed system would be better I think. I made this original chart, yet I have no ability to maintain it. Now I watch it fall into neglect and disrepair while useful PRs like yours and important bug fixes like #9127 gather dust 😕

File: stable/efs-provisioner/values.yaml [hunk 1]
```diff
@@ -54,6 +54,19 @@ serviceAccount:
...
+podAnnotations: {}
+  # iam.amazonaws.com/role: efs-provisioner-role
+
+## Node labels for pod assignment
+##
+nodeSelector: {}
+
+# Affinity for pod assignment
+# Ref: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#affinity-and-anti-affinity
+# affinity: {}
```
Comment (unguiculus) [inline stable/efs-provisioner/values.yaml]:
```suggestion
affinity: {}
```

File: stable/efs-provisioner/README.md [hunk 2]
```diff
@@ -92,6 +92,19 @@ rbac:
...
+podAnnotations: {}
+  # iam.amazonaws.com/role: efs-provisioner-role
+
+## Node labels for pod assignment
+##
+nodeSelector: {}
+
+# Affinity for pod assignment
+# Ref: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#affinity-and-anti-affinity
+# affinity: {}
```
Comment (unguiculus) [inline stable/efs-provisioner/README.md]:
```suggestion
affinity: {}
```

**Review by unguiculus — CHANGES_REQUESTED** [review]

File: stable/efs-provisioner/values.yaml (hunk 1 above)
Comment (SweetOps) [reply stable/efs-provisioner/values.yaml]: has been uncommented as you requested

File: stable/efs-provisioner/README.md (hunk 2 above)
Comment (SweetOps) [reply stable/efs-provisioner/README.md]: has been uncommented as you requested

**Comment by SweetOps** [comment]
@unguiculus , please update your review

**Review by unguiculus — APPROVED** [review]

**Comment by whereisaaron** [comment]
This merge appears to introduce two `annotations` sections to the Pod template,  where `.Value.annotations` will override `.Values.podAnnotations`, see #13494.

**Pull request merged**